lib/index.rb

```diff
@@ -84,6 +84,18 @@ def with_lock
       end
     end
 
+    def sync_mappings
+      errors = {}
```

Comment:
The Rubyist in me would write this as:

```ruby
{}.tap do |errors|
  mappings.each do |type, mapping|
    # ...
  end
end
```

Saves you from having to declare and return `errors`!